### PR TITLE
fix(operator yaml): update deployment upgrade strategy as Recreate

### PR DESCRIPTION
**Pull request by mittachaitu**
Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>
This PR updates the control plane deployment upgrade strategy to `Recreate`.
When we set the upgrade strategy to `Recreate` deployment controller will
make sure during upgrade time(patching pod template spec) after deleting old pod
new pod with the changes will come to running state.

### CODE REVIEW

File: k8s/charts/openebs/templates/deployment-local-provisioner.yaml [hunk 1]
```diff
@@ -12,6 +12,9 @@ metadata:
     openebs.io/version: {{ .Values.release.version }}
 spec:
   replicas: {{ .Values.provisioner.replicas }}
+  strategy:
+    type: {{ .Values.provisioner.upgradeStrategy }}
```
Comment (kmova) [inline k8s/charts/openebs/templates/deployment-local-provisioner.yaml]: Can this be changed to local PV provisioner. Also update the above replicas to use the localpvprovisioner (this has been fixed in upstream)

File: k8s/charts/openebs/templates/deployment-admission-server.yaml [hunk 2]
```diff
@@ -12,6 +12,9 @@ metadata:
     openebs.io/version: {{ .Values.release.version }}
 spec:
   replicas: {{ .Values.webhook.replicas }}
+  strategy:
+    type: {{ .Values.webhook.upgradeStrategy }}
+    rollingUpdate: null
```
Comment (kmova) [inline k8s/charts/openebs/templates/deployment-admission-server.yaml]: should rollingUpdate be set to null .. if the upgradeStrategy is set to  `Recreate`. Having said that, should we even give this option to select the upgradeStrategy or hardcode it to "Recreate" ?

File: k8s/charts/openebs/templates/deployment-local-provisioner.yaml (hunk 1 above)
Comment (mittachaitu) [reply k8s/charts/openebs/templates/deployment-local-provisioner.yaml]: I will update it

File: k8s/charts/openebs/templates/deployment-admission-server.yaml (hunk 2 above)
Comment (mittachaitu) [reply k8s/charts/openebs/templates/deployment-admission-server.yaml]: We will hardcode it to `Recreate`

**Pull request merged**